Orange/widgets/visualize/owscatterplot.py

```diff
@@ -394,18 +395,40 @@ def set_subset_data(self, subset_data):
 
     # called when all signals are received, so the graph is updated only once
     def handleNewSignals(self):
-        self.graph.new_data(self.data_metas_X, self.subset_data)
+        if self.data is None or (self.data and not sp.issparse(self.data.X)):#  table.issparse() !!!
```

Comment:
Or maybe even better:
```
if self.data is not None and self.data.is_sparse():
    self.sparse_to_dense()
else:
    self.graph.new_data(self.data_metas_X, self.subset_data)
```

Doing so you also don't need that if at the beginning of `sparse_to_dense` method.